src/hierarchical_deterministic/bip39/bip39_entropy.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{decl_secret_bytes, prelude::*, UniffiCustomTypeConverter};
+
+macro_rules! entropy_with_byte_counts {
+    (
+        $(
+            #[doc = $expr: expr]
+        )*
+        $enum_name: ident:
+        $(
+            $byte_count: literal,
+        )+
+    ) => {
+        paste! {
+            $(
+                decl_secret_bytes!(
+                    [< Entropy $byte_count Bytes >],
+                    $byte_count
+                );
+            )+
+
+            $(
+                #[doc = $expr]
+            )*
+            #[derive(Zeroize, ZeroizeOnDrop)]
+            pub enum $enum_name {
```

Comment:
here the enum, named `BIP39Entropy` gets declared, and repeating again the `16, 20, 24, 28, 32, ` to create the 5 cases/variants: `EntropyOf16Bytes(Entropy16Bytes)`, `EntropyOf20Bytes(Entropy20Bytes)` etc.

Reply:
The enum allows us to work ONLY with fixed size arrays! Which is quite nice!